code/addons/test/src/preset.ts

```diff
@@ -102,6 +116,65 @@ export const experimental_serverChannel = async (channel: Channel, options: Opti
         addonVersion,
       });
     });
+
+    store.onStateChange(async (state, previous) => {
+      if (state.watching && !previous.watching) {
+        await telemetry('testing-module-watch-mode', {
+          provider: TEST_PROVIDER_ID,
+          watchMode: state.watching,
+        });
+      }
+    });
+
+    channel.on(
+      TESTING_MODULE_PROGRESS_REPORT,
+      async (payload: TestingModuleProgressReportPayload) => {
+        if (payload.providerId !== TEST_PROVIDER_ID) {
```

Comment:
Can you elaborate why a separate telemetry handling is necessary for addon test?

Reply:
Because we want telemetry to include the information of whether or not coverage/a11y was enabled for the run. And that information is now completely opaque to SB core, so the default core telemetry can't include that information.